packages/remix-dev/package.json

```diff
@@ -68,9 +68,22 @@
     "msw": "^0.39.2",
     "shelljs": "^0.8.5",
     "tiny-invariant": "^1.2.0",
-    "type-fest": "^2.12.2"
+    "type-fest": "^2.12.2",
+    "@yarnpkg/esbuild-plugin-pnp": "^2.0.0"
   },
   "engines": {
     "node": ">=14"
+  },
+  "peerDependencies": {
+    "@remix-run/serve": "1.5.1",
+    "@yarnpkg/esbuild-plugin-pnp": "^2.0.0"
+  },
+  "peerDependenciesMeta": {
+    "@remix-run/serve": {
+      "optional": true
+    },
+    "@yarnpkg/esbuild-plugin-pnp": {
+      "optional": true
+    }
```

Comment:
I'm not convinced yet that having `@yarnpkg/esbuild-plugin-pnp` as optional dependency is the way we want to go. I'm leaning towards "yes", but will discuss with the team. If we do go with this optional dependency, then we'd need some docs that explain to users that they need to install it if they are using yarn pnp.